osu.Server.Queues.ScoreStatisticsProcessor/Processors/PerformanceProcessor.cs

```diff
@@ -66,9 +71,40 @@ private double computePerformance(Ruleset ruleset, Mod[] mods, ScoreInfo score,
                     ModValue = (uint)legacyModValue
                 }, transaction).ToArray();
 
-            var difficultyAttributes = rawDifficultyAttribs.ToDictionary(a => (int)a.attrib_id).Map(score.RulesetID, beatmap);
-            var performanceCalculator = ruleset.CreatePerformanceCalculator(difficultyAttributes, score);
-            return performanceCalculator.Calculate();
+            DifficultyAttributes attributes;
+
+            switch (score.RulesetID)
+            {
+                case 0:
+                    // Some attributes aren't databased because they're present in other tables.
+                    attributes = new OsuDifficultyAttributes
+                    {
+                        DrainRate = beatmap.diff_drain,
+                        HitCircleCount = beatmap.countNormal,
+                        SliderCount = beatmap.countSlider,
+                        SpinnerCount = beatmap.countSpinner
+                    };
```

Comment:
This is a little unfortunate. I'm all for making another class to hold these to be passed into `FromDifficultyAttributes()`, but please suggest a class name... `ExtraBeatmapDifficultyAttributes`?